cronjob_scripts/trigger-package-build.py

```diff
@@ -41,10 +42,10 @@ def trigger_for_arch(target_arch: str):
     tracking_worktree_path = checkout_worktree_for_arch(target_arch)
     excludes = get_excludes(tracking_worktree_path, days=7, max_failures=5)
 
-    popular_crates = get_popular_crates()
-    requested_crates = get_requested_crates(period="1 day", arch=target_arch)
+    possible_crates = set(get_crates_io_popular_crates())
```

Comment:
this now downloads 700MB from crates.io every time we want to check crates (every hour + anytime I want to smoke test anything locally), with no cache.

I was thinking about this the other day, and we could potentially only extract the files we're planning to ingest, and save it to somewhere in /tmp, with a cache key based on today's date or something?

Reply:
sorry... once per arch because there get_crates_io_popular_crates is not wrapped in @functools.lru_cache

Reply:
Yes according to [polars user guide](https://docs.pola.rs/user-guide/io/parquet/#)

> Loading or writing [Parquet files](https://parquet.apache.org/) is lightning fast as the layout of data in a Polars DataFrame in memory mirrors the layout of a Parquet file on disk in many respects.

And there's an API for doing it, with optional compression.

https://docs.pola.rs/api/python/stable/reference/api/polars.DataFrame.write_parquet.html#polars.DataFrame.write_parquet

If we are going to cache stuff and current progress in trigger/* branch, then this will not be necessary.

Reply:
I've added caching, the file to be cached is only 2785 B large, much smaller than the 700MB file @alsuren